# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/new-regexp/regexp-shim.cc
# Commit: c474eb63f626
# Full Hash: c474eb63f626923dc192a65f4f144a280f544fe2
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-05-13 21:46:27
# Regressor Bug: 1634135
# File Overlap Count: 1
# Description:
#   Bug 1634135: Disable named capture parsing until fully supported r=mgaudet
#   
#   The engine supports parsing named captures, but we don't have the code in place to expose the captured groups. Until we do, this code will make sure that we get a syntax error when parsing them.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73115
# ==============================================================================

diff -r e0f929077cce -r c474eb63f626 js/src/new-regexp/regexp-shim.cc
--- a/js/src/new-regexp/regexp-shim.cc	Wed May 13 16:19:24 2020 +0000
+++ b/js/src/new-regexp/regexp-shim.cc	Wed May 13 16:19:57 2020 +0000
@@ -183,7 +183,7 @@
 
 Handle<FixedArray> Isolate::NewFixedArray(int length) {
   MOZ_RELEASE_ASSERT(length >= 0);
-  MOZ_CRASH("TODO");
+  return Handle<FixedArray>::fromHandleValue(JS::UndefinedHandleValue);
 }
 
 template <typename CharT>
